src/main/drivers/compass/compass_hmc5883l.c

```diff
@@ -145,80 +153,121 @@ static void hmc5883_extiHandler(extiCallbackRec_t* cb)
 }
 #endif
 
-static void hmc5883lConfigureDataReadyInterruptHandling(void)
+static void hmc5883lConfigureDataReadyInterruptHandling(magDev_t* mag)
 {
 #ifdef USE_MAG_DATA_READY_SIGNAL
-
-    if (!(hmc5883InterruptIO)) {
+    if (mag->magIntExtiTag == IO_TAG_NONE) {
         return;
     }
+
+    const IO_t magIntIO = IOGetByTag(mag->magIntExtiTag);
+
 #ifdef ENSURE_MAG_DATA_READY_IS_HIGH
-    uint8_t status = IORead(hmc5883InterruptIO);
+    uint8_t status = IORead(magIntIO);
     if (!status) {
         return;
     }
 #endif
 
-    EXTIHandlerInit(&hmc5883_extiCallbackRec, hmc5883_extiHandler);
-    EXTIConfig(hmc5883InterruptIO, &hmc5883_extiCallbackRec, NVIC_PRIO_MAG_INT_EXTI, EXTI_Trigger_Rising);
-    EXTIEnable(hmc5883InterruptIO, true);
+#if defined (STM32F7)
+    IOInit(magIntIO, OWNER_COMPASS_EXTI, 0);
+    EXTIHandlerInit(&mag->exti, hmc5883_extiHandler);
+    EXTIConfig(magIntIO, &gmag->exti, NVIC_PRIO_MPU_INT_EXTI, IO_CONFIG(GPIO_MODE_INPUT,0,GPIO_NOPULL));
+#else
+    IOInit(magIntIO, OWNER_COMPASS_EXTI, 0);
+    IOConfigGPIO(magIntIO, IOCFG_IN_FLOATING);
+    EXTIHandlerInit(&mag->exti, hmc5883_extiHandler);
+    EXTIConfig(magIntIO, &mag->exti, NVIC_PRIO_MAG_INT_EXTI, EXTI_Trigger_Rising);
+    EXTIEnable(magIntIO, true);
+#endif
+#else
+    UNUSED(mag);
 #endif
 }
 
-static bool hmc5883lRead(int16_t *magData)
-{
-    uint8_t buf[6];
 #ifdef USE_MAG_SPI_HMC5883
-    bool ack = hmc5883SpiReadCommand(MAG_DATA_REGISTER, 6, buf);
-#else
-    bool ack = i2cRead(MAG_I2C_INSTANCE, MAG_ADDRESS, MAG_DATA_REGISTER, 6, buf);
+/*
+ * XXX Fixme
+ * HMC5983 datasheet states SPI mode is
+ *     SCK is high when CS is high
+ *     Data is sampled at the rising edge of SCK
+ * However, it seems that SCK condition is ignored (works with SCK low when CS is high).
```

Comment:
SPI slave should ignore SCK as long as CS is not asserted. 
Only SCK state when asserting CS may need some consideration